src/theory/arith/theory_arith.cpp

```diff
@@ -164,19 +164,25 @@ void TheoryArith::postCheck(Effort level)
 {
   d_im.reset();
   Trace("arith-check") << "TheoryArith::postCheck " << level << std::endl;
+  if (Theory::fullEffort(level))
```

Comment:
Can this be moved to line 202?

Reply:
I don't think so. Note that there are multiple ways to return between line 167 and line 202.